services/src/main/scala/cromwell/services/keyvalue/impl/KeyValueServiceActor.scala

```diff
@@ -31,14 +29,27 @@ case class KeyValueServiceActor(serviceConfig: Config, globalConfig: Config) ext
     }
   }
 
-  private def doPut(put: KvPut): Future[KvResponse] = Future.successful {
-    store += put.pair.key -> put.pair.value
-    KvPutSuccess(put)
+  private def doPut(put: KvPut): Future[KvResponse] = {
+    put.pair.value match {
+      case Some(backendVal) => updateBackendKeyValuePair(put.pair.key.workflowId,
+        put.pair.key.jobKey,
+        put.pair.key.key,
+        put.pair.value.get).map(_ => KvPutSuccess(put))
+      case None => Future(KvFailure(put, new Throwable(s"Failed to find the value associated to key: ${put.pair.key.key}. This key cannot be added to the BackendKVStore.")))
+    }
   }
 
-  private def doGet(get: KvGet): Future[KvResponse] = Future.successful {
-    store.get(get.key) match {
-      case Some(maybeValue) => KvPair(get.key, maybeValue)
+  private def doGet(get: KvGet): Future[KvResponse] = {
+    val executionInfo = getBackendValueByKey(
```

Comment:
How about something like `val backendValue` instead of `val executionInfo`?

`executionInfo` reminds me of the olde worlde table "EXECUTION_INFO".